src/Mapper/Entry.php

```diff
@@ -80,16 +94,65 @@ private function buildFields(ResourceContentType $contentType, array $fields, $l
     }
 
     /**
-     * @param ResourceContentTypeField $fieldConfig
-     * @param array                    $fieldData
+     * If a previous entry is give, this methods will combine all values from the $fields
+     * array an those in the already built entry.
+     * This is done so in the event of partial resource building (i.e. after having queried
+     * the API using the "select" operator), the resulting object is increasingly updated,
+     * therefore the new object will contain the biggest set of old plus new fields.
+     *
+     * @param ResourceContentType $contentType
+     * @param array               $fields      The field values that have been returned by the API
+     * @param string|null         $locale      Either a string if a locale is used, or null if all locales are present
+     * @param ResourceClass       $previous    the previous entry object that was already built, if present
+     *
+     * @return array
+     */
+    private function mergePreviousFields(
+        ResourceContentType $contentType,
+        array $fields,
+        $locale,
+        ResourceClass $previous
+    ) {
+        $environment = $previous->getEnvironment();
+        $defaultLocale = $environment->getDefaultLocale()->getCode();
+
+        // If no locale is available, it means we need to deal with all of them.
+        // To normalize this behavior, we treat them as an array either way.
+        $locales = $locale
+            ? [$locale]
+            : \array_map(function (ResourceLocale $locale) {
+                return $locale->getCode();
+            }, $environment->getLocales());
+
+        // As we can't know for certain which fields will be present (some might not be available
+        // in the current API response, some might be excluded anyway because they're blank...)
+        // we just use the field definitions from the content type and progressively check which fields
+        // are actually present.
+        foreach ($contentType->getFields() as $field) {
+            $fieldId = $field->getId();
+            $fieldLocales = $field->isLocalized() ? $locales : [$defaultLocale];
+
+            foreach ($fieldLocales as $locale) {
+                if (!isset($fields[$fieldId][$locale]) && $previous->has($fieldId)) {
```

Comment:
Is it possible to ensure with a revision check that the object you're updating has not been changed?